File: ts/wam/4692.ts
export const WamChannelCoreEvent = 4692 as const

export interface ChannelCoreEventEvent {
  /** field 1, wire `channel_core_event_type` */
  channelCoreEventType?: CHANNELEVENTTYPE
  /** field 2, wire `channel_entry_point` */
  channelEntryPoint?: CHANNELENTRYPOINT
  /** field 3, wire `channel_entry_point_app` */
  channelEntryPointApp?: CHANNELENTRYPOINTAPP
  /** field 4, wire `cid` */
  cid?: string
  /** field 5, wire `entry_point_metadata` */
  entryPointMetadata?: string
  /** field 6, wire `event_surface` */
  eventSurface?: CHANNELEVENTSURFACE
  /** field 7, wire `channel_directory_session_id` */
  channelDirectorySessionId?: number
  /** field 8, wire `channel_core_event_sequence_number` */
  channelCoreEventSequenceNumber?: number
  /** field 9, wire `directory_channel_index` */
  directoryChannelIndex?: number
  /** field 10, wire `channel_entry_point_metadata` */
  channelEntryPointMetadata?: CHANNELENTRYPOINTMETADATA
  /** field 12, wire `channel_event_unit` */
  channelEventUnit?: CHANNELEVENTUNIT
  /** field 13, wire `similar_channels_session_id` */
  similarChannelsSessionId?: number
  /** field 14, wire `discovery_surface` */
  discoverySurface?: TSSURFACE
  /** field 15, wire `channel_discovery_query_id` */
  channelDiscoveryQueryId?: string
  /** field 16, wire `channel_discovery_search_id` */
  channelDiscoverySearchId?: string
  /** field 17, wire `unified_session_id` */
  unifiedSessionId?: string
  /** field 18, wire `updates_tab_session_id` */
  updatesTabSessionId?: number
  /** field 19, wire `channel_request_metadata` */
  channelRequestMetadata?: string
  /** field 20, wire `trace_id_int` */
  traceIdInt?: number
  /** field 21, wire `core_event_trigger` */
  coreEventTrigger?: COREEVENTTRIGGERTYPE
}

File: go/wam/4692.go
const WamChannelCoreEvent = 4692 // channel: regular

type ChannelCoreEventEvent struct {
	ChannelCoreEventType CHANNELEVENTTYPE `wam:"channel_core_event_type"` // field 1
	ChannelEntryPoint CHANNELENTRYPOINT `wam:"channel_entry_point"` // field 2
	ChannelEntryPointApp CHANNELENTRYPOINTAPP `wam:"channel_entry_point_app"` // field 3
	Cid string `wam:"cid"` // field 4
	EntryPointMetadata string `wam:"entry_point_metadata"` // field 5
	EventSurface CHANNELEVENTSURFACE `wam:"event_surface"` // field 6
	ChannelDirectorySessionId int64 `wam:"channel_directory_session_id"` // field 7
	ChannelCoreEventSequenceNumber int64 `wam:"channel_core_event_sequence_number"` // field 8
	DirectoryChannelIndex int64 `wam:"directory_channel_index"` // field 9
	ChannelEntryPointMetadata CHANNELENTRYPOINTMETADATA `wam:"channel_entry_point_metadata"` // field 10
	ChannelEventUnit CHANNELEVENTUNIT `wam:"channel_event_unit"` // field 12
	SimilarChannelsSessionId int64 `wam:"similar_channels_session_id"` // field 13
	DiscoverySurface TSSURFACE `wam:"discovery_surface"` // field 14
	ChannelDiscoveryQueryId string `wam:"channel_discovery_query_id"` // field 15
	ChannelDiscoverySearchId string `wam:"channel_discovery_search_id"` // field 16
	UnifiedSessionId string `wam:"unified_session_id"` // field 17
	UpdatesTabSessionId int64 `wam:"updates_tab_session_id"` // field 18
	ChannelRequestMetadata string `wam:"channel_request_metadata"` // field 19
	TraceIdInt int64 `wam:"trace_id_int"` // field 20
	CoreEventTrigger COREEVENTTRIGGERTYPE `wam:"core_event_trigger"` // field 21
}